### DISCUSSION TITLE
S3에 저장된 여러 해상도 사진들의 네이밍 컨벤션

### DISCUSSION THREAD

**Discussion by WaiNaat** [woowacourse-teams/2023-pium#384] [개발 논의]
## 참여자

쵸파, 클린, 참새

## 문제 상황

S3에 저장된 사전 식물들의 사진을 저해상도로 제공할 계획입니다 (#381)
현재 원본 이미지는 `https://static.pium.life/dict/{file_name}.jpg` 경로로 접근이 가능한데, 새로운 사진들은 어떤 경로로 접근해야 할까요?

정적 파일이고 백엔드 측에서는 새로운 저화질 사진들의 url을 알지 못하기 때문에 원본 이미지로부터 파싱을 해서 이미지를 요청해야 합니다.

### 1안

`https://static.pium.life/dict/small/webp/{file_name}.webp`

- 원본 사진 주소를 파싱해서 새로운 주소를 얻는 과정이 복잡해진다
- 폴더 구조가 너무 깊어진다

### 2안

`https://static.pium.life/dict/{file_name}.small.png`

- `.`을 통해서 속성들을 명시
- 상대적으로 파싱 난이도가 쉽다.

## 결론

2안의 구조를 따르기로 했습니다.
현재 지원하는 타입들은 다음과 같아요.
```
https://static.pium.life/dict/{file_name}.{nickname}.{extension}

{nickname} -> 'small' | 'x-small'
{extension} -> 'png' | 'webp'
```

**Comment by hozzijeong** [comment]
[이름][사이즈][확장자]의 네이밍 규칙 멋져요 👍

**Comment by bassyu** [comment]
확인했습니다! 이미지 최적화 수작업 감동이에요 👍👍

**Comment by WaiNaat** [comment]
적용 완료
#387 
https://github.com/pium-official/resize-image
